src/snmp_core.cc

```diff
@@ -743,18 +743,14 @@ peer_Inst(oid * name, snint * len, mib_tree_entry * current, oid_ParseFn ** Fn)
         instance[*len] = 1 ;
         *len += 1;
     } else {
-        int no = name[current->len] ;
-        int i;
-        // Note: This works because the Config.peers keeps its index according to its position.
-        for ( i=0 ; peers && (i < no) ; peers = peers->next, ++i ) ;
-
-        if (peers) {
-            debugs(49, 6, "snmp peer_Inst: Encode peer #" << i);
+        const auto no = name[current->len];
```

Comment:
Please do keep the old `int`-based declaration, the important (but wrong) source code comment, and the (refactored) silly loop above. This change makes `no` a `char`. That is worse than `int` because we do not know whether `char` is signed. There was a recent security vulnerability because of a similar conversion that went wrong. Also, we increment `no` below. Also, this official code is probably wrong because it should use `CachePeer::index` instead. Let's not bring all those concerns into this PR scope.